incubator/hnc/third_party/open-policy-agent/gatekeeper/pkg/webhook/certs.go

```diff
@@ -517,13 +564,53 @@ func (r *ReconcileVWH) Reconcile(request reconcile.Request) (reconcile.Result, e
 		if err := r.client.Update(r.ctx, vwh); err != nil {
 			return reconcile.Result{Requeue: true}, err
 		}
+
+		// Ensure certs on CRDs.
+		if result, err := r.ensureCRDCerts(artifacts.CertPEM); err != nil {
```

Comment:
Can you help explain the logic to me here. As I see it, we use all crds in `r.crdNames`, initialize it as unstructured data, and then loop trying to inject the CA bundle. However, if ANY of the CRDs do not have the `webhookClientConfig` field (https://github.com/kubernetes-sigs/multi-tenancy/pull/953/files#diff-5561b1b3c7de2f4fc72e62d0f20c6e97R238) we return an error from the loop, meaning we would never reach `wasCAInjected = true` below. Moreover, if the first CRD has an error, we don't even attempt the others.

I guess where my concern is that `crdNames` is a slice of strings, and we don't have any real validation. If it doesn't have a field we care about, we reject they entire CA injection. 

This is likely not a problem today, but I see it as a future bug potential, especially if it means some webhooks may not get their certs.

Reply:
Right, the review from https://github.com/open-policy-agent/cert-controller/pull/7 had the same concern. 

I updated the logic to only return (and only return error) from the loop for transient errors, e.g. cannot get or update, so the request will be re-queued to finish the loop.

For CRD not found, being deleted, or not having caBundle field, we will just have a log message and continue, to eventually return no error. Because we don't need to retry for these cases.